common-custom-user-data-maven-extension/src/main/java/com/gradle/maven/common/user/data/extension/GroovyScriptUserData.java

```diff
@@ -8,11 +8,16 @@
 import org.apache.maven.monitor.logging.DefaultLog;
 import org.codehaus.plexus.logging.Logger;
 
+import javax.annotation.Nullable;
 import java.io.File;
 
 final class GroovyScriptUserData {
 
-    static void evaluate(MavenSession session, GradleEnterpriseApi gradleEnterprise, Logger logger) throws MavenExecutionException {
+    static void evaluate(@Nullable MavenSession session, GradleEnterpriseApi gradleEnterprise, Logger logger) throws MavenExecutionException {
+        if (session == null) {
+            logger.warn("Cannot evaluate groovy script when the Maven session in not available");
```

Comment:
Should we instead document `session` and `project` as nullable and evaluate it anyway?

Reply:
not sure what you mean. If the session is not available (very edge case scenarios with very early Maven build failures), we cannot grab the location of the groovy script (from `session.getRequest().getMultiModuleProjectDirectory()` +`.mvn/gradle-enterprise-custom-user-data.groovy`)

Reply:
moot